native/src/main/java/io/ballerina/lib/redis/utils/ConversionUtils.java

```diff
@@ -180,6 +180,44 @@ public static BArray createBStringArrayFromBMap(BMap<BString, Object> bMap) {
     public static BError createBError(Throwable e) {
         return ErrorCreator.createError(ModuleUtils.getModule(), Constants.REDIS_ERROR_TYPE,
                 StringUtils.fromString(e.getMessage()),
-                ErrorCreator.createError(StringUtils.fromString(e.getMessage()), e), null);
+                e.getCause() != null ? ErrorCreator.createError(e.getCause()) : null, null);
+    }
+
+    /**
+     * Returns the map value from a Ballerina map for a given key or, null if a non-map value is found.
+     *
+     * @param map the Ballerina map value
+     * @param key the key
+     * @return the map value or null
+     */
+    public static BMap<BString, Object> getMapValueOrNull(BMap<BString, Object> map, BString key) {
+        if (map == null) {
+            return null;
```

Comment:
Can't we return an empty map rather than null ? Returning `null` is an antipattern in java coding standards specially when there is a collection.

Reply:
Yes. But in this case we'll end up with a lot more additional checks + empty objects in the heap if we replace null values with empty objects related to configs. As per what we agreed during the offline review, shall we stick to the existing approach in here?